packages/cletus/src/operations/architect.tsx

```diff
@@ -311,3 +315,599 @@ export const type_create = operationOf<
+            // Notify processor that data is available
+            if (queueHasData) {
+              const resolver = queueHasData;
+              queueHasData = null;
+              resolver();
+            }
+            
+          } catch (error) {
+            log(`Warning: Failed to process file ${file.file}: ${(error as Error).message}`);
+          }
+        }
+      } catch (error) {
+        parsingError = error as Error;
+      } finally {
+        parsingComplete = true;
+        // Wake up processor if it's waiting
+        if (queueHasData) {
+          const resolver = queueHasData;
+          queueHasData = null;
+          resolver();
+        }
+      }
+    })();
+    
+    // Processor promise - consumes queue and calls AI
+    const processorPromise = (async () => {
+      let filesProcessed = 0;
+      let currentBatch = '';
+      let currentBatchFiles: string[] = [];
+      
+      while (true) {
+        // Wait for data if queue is empty and parsing is ongoing
+        while (dataQueue.length === 0 && !parsingComplete) {
+          await new Promise<void>(resolve => {
+            queueHasData = resolve;
+          });
+        }
+        
+        // Exit if queue is empty and parsing is done
+        if (dataQueue.length === 0 && parsingComplete) {
+          break;
+        }
+        
+        // Process available data
+        while (dataQueue.length > 0) {
+          const fileData = dataQueue.shift()!;
+          queuedSize -= fileData.text.length;
+          
+          // Notify parser that space is available
+          if (queueHasSpace && queuedSize < maxQueueSize) {
+            const resolver = queueHasSpace;
+            queueHasSpace = null;
+            resolver();
+          }
+          
+          const fileContent = `\n\n=== File: ${fileData.file} ===\n${fileData.text}`;
+          
+          // Check if adding this file would exceed batch size
+          if (currentBatch.length > 0 && currentBatch.length + fileContent.length > TYPE_IMPORT_CONSTS.BATCH_SIZE) {
+            // Process current batch
+            chatStatus(`Processing batch (${currentBatchFiles.length} files, ${currentBatch.length} chars)...`);
+            
+            try {
+              await extractor.get('result', { text: currentBatch }, ctx);
+            } catch (error) {
+              log(`Warning: Failed to process batch: ${(error as Error).message}`);
+            }
+            
+            // Start new batch
+            currentBatch = fileContent;
+            currentBatchFiles = [fileData.file];
+          } else {
+            currentBatch += fileContent;
+            currentBatchFiles.push(fileData.file);
+          }
+          
+          filesProcessed++;
+          chatStatus(`Processed ${filesProcessed}/${importableFiles.length} files`);
+          
+          // Check if we've reached max types (if specified)
+          if (max && discoveredTypes.size >= max && (!hints || hints.length === 0)) {
```

Comment:
We shouldn't stop here. On add discovered type tool we should throw an error or respond with a message when a type can't be added because the max is hit and another one that has less instances should be removed or don't request to add the type again. Actually there is an applicable function you can add to tools that determine whether they are even an option in the prompt. The add discovered type should only be available if the max hasnt been hit and some of the other tools should only be applicable that deal with discovered types of there are any discovered types.